docs/release-notes/self-hosted/self-hosted-v12.0.0.md

```diff
@@ -0,0 +1,161 @@
+
+1.  Follow the instructions to [upgrade your Codacy Self-hosted instance](https://docs.codacy.com/v12.0/chart/maintenance/upgrade/).
+
+1.  Update your Codacy command-line tools to the following versions:
+
+    -   [Codacy Analysis CLI 7.8.3](https://github.com/codacy/codacy-analysis-cli/releases/tag/7.8.3)
+    -   [Codacy Coverage Reporter 13.13.1](https://github.com/codacy/codacy-coverage-reporter/releases/tag/13.13.1)
+
+## Breaking changes
+
+-   If you're updating Kubernetes to v1.25, upgrade to Codacy Self-hosted v12.0.0 before doing so, as some Kubernetes endpoints have been deprecated and may lead to service outage if used with earlier versions of Codacy Self-hosted.
+
+-   If using the monitoring functionality [Grafana + Prometheus + Loki](https://docs.codacy.com/v12.0/chart/configuration/monitoring/#setting-up-monitoring-using-grafana-prometheus-and-loki), ensure Pod Security Policies are disabled for Loki and Promtail by upgrading in the following order:
+
+    1.  Upgrade to Codacy Self-hosted v12.0.0
+
+    1.  Update Loki and Promtail by following the respective installation instructions ([Loki](https://docs.codacy.com/v12.0/chart/configuration/monitoring/#2-installing-loki), [Promtail](https://docs.codacy.com/v12.0/chart/configuration/monitoring/#3-installing-promtail))
+
+    1.  Upgrade Kubernetes to v1.25
+
+-   If you have set `scheduler.enable = "true"` in `values.yaml`, do the following before migrating to Codacy Self-hosted v12.0.0.
+
+    Assuming you're using the default `codacy` namespace:
+
+    1.  Remove all `scheduler.*` settings from `values.yaml`
+
+    1.  Update the ConfigMap of `worker-manager`:
+
+        ```bash
+        kubectl patch configmap/worker-manager --namespace codacy --type merge --patch '{"data":{"CONFIG_FORCE_codacy_kubernetes_scheduler_enable": "false","CONFIG_FORCE_codacy_kubernetes_scheduler_name":"default-scheduler"}}'
+        ```
+
+    1.  Restart the `worker-manager` Deployment:
+
+        ```bash
+        kubectl rollout restart deployment/worker-manager -n codacy
+        ```
+
+    1.  List any pending `worker` Pods:
+
+        ```bash
+        kubectl get pods -l app=worker --field-selector=status.phase=Pending -n codacy
+        ```
+
+    1.  If there are any pending `worker` Pods, remove them:
+
+        !!! warning
+            This is a destructive action. Make sure you understand the consequences.
+
+        ```bash
+        kubectl delete pods -l app=worker --field-selector=status.phase=Pending -n codacy
+        ```
+
+## Product enhancements
+
+-   Added support for [Kubernetes 1.25.\*](https://docs.codacy.com/v12.0/chart/requirements/#kubernetes-or-microk8s-cluster-setup) (REL-1306)
+-   Added support for [<span class="skip-vale">cgroup</span> v2](https://kubernetes.io/docs/concepts/architecture/cgroups/). (<!-- TODO is there a task tracking this activity?-->)
```

Comment:
@lolgab I'm having trouble finding the relevant task. Was this done as part of another task, or maybe not tracked on Jira?

Thanks! 🙏 

```suggestion
-   Added support for [<span class="skip-vale">cgroup</span> v2](https://kubernetes.io/docs/concepts/architecture/cgroups/). (<!-- TODO is there a task tracking this activity?-->)
```

Reply:
[After confirming](https://github.com/codacy/docs/pull/1767#issuecomment-1630853902) with @lolgab, we can actually skip adding this item to the release notes:

```suggestion
```